hivelink-core/src/main/java/org/apache/iceberg/hivelink/core/schema/MergeHiveSchemaWithAvro.java

```diff
@@ -97,12 +98,12 @@ public Schema.Field field(String name, TypeInfo field, Schema.Field partner, Sch
    * If the schema is not an option schema or if there is no default value, schema is returned as-is
    */
   private Schema reorderOptionIfRequired(Schema schema, Object defaultValue) {
-    if (AvroSchemaUtil.isOptionSchema(schema) && defaultValue != null) {
-      boolean isNullFirstOption = schema.getTypes().get(0).getType() == Schema.Type.NULL;
-      if (isNullFirstOption && defaultValue.equals(JsonProperties.NULL_VALUE)) {
-        return schema;
+    if (AvroSchemaUtil.isOptionSchema(schema)) {
+      int nullIndex = AvroSchemaUtil.getNullIndex(schema);
+      if (defaultValue != null && !defaultValue.equals(JsonProperties.NULL_VALUE)) {
+        return Schema.createUnion(schema.getTypes().get(1 - nullIndex), Schema.create(Type.NULL));
       } else {
-        return Schema.createUnion(schema.getTypes().get(1), schema.getTypes().get(0));
+        return Schema.createUnion(Schema.create(Type.NULL), schema.getTypes().get(1 - nullIndex));
```

Comment:
If no default value is specified, why do we need to reorder. The schema is valid as-is. The original intention of the method was to reorder to match the provided data type to work around invalid defaults.